shared/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -316,7 +322,55 @@ namespace datadog::shared::nativeloader
             return E_FAIL;
         }
 
-        // Guard rails have all passed, so we enable (and flush) logs if necessary
+        // Workload Selection
+        #if defined(BUILD_WITH_WORKLOAD_SELECTION)
+        if (IsSingleStepInstrumentation() && !IsRunningOnIIS()) {
+          enum class InjectionStatus : uint8_t {
+            ALLOW,
+            DENY,
+            UNKNOWN
+          } injection_status = InjectionStatus::UNKNOWN;
+
+          plcs::init();
+          plcs::set_params(plcs::StringEvaluator::PROCESS_EXE, ToString(process_name));
+          plcs::set_params(plcs::StringEvaluator::RUNTIME_LANGUAGE, "dotnet");
+
+          // TODO: Remove once the policy engine will be able to log the context.
+          Log::Info("CorProfiler::Initialize: Workload Selection Context: \"process.name:", process_name, " runtime:dotnet\"");
+
+          plcs::register_action(plcs::Action::INJECT_ALLOW, [&injection_status](plcs::Result eval_result, const std::vector<const char*>&, const char* desc) -> std::optional<plcs::Error> {
+              if (injection_status == InjectionStatus::ALLOW) return std::nullopt;
+              if (eval_result == plcs::Result::ALLOW)
```

Comment:
Discussed offline, let's use another name than Allow for the result to be consistent with the C implem and avoid the confusion with the action name.